### PR TITLE
Refactor logic around snapper boot into snapshot

### PR DESCRIPTION
**Pull request by asmorodskyi**

### CODE REVIEW

File: tests/installation/grub_test.pm [hunk 1]
```diff
@@ -63,24 +63,7 @@ sub run() {
...
-        send_key 'ret';
-        # bsc#956046  check if we are in first menu-entry, or not
-        if (check_screen("boot-menu-snapshot-bootmenu")) {
-            record_soft_failure 'bsc#956046';
-            send_key 'down', 1;
-            save_screenshot;
-        }
-        send_key 'ret';
-    }
+    boot_into_snapshot if (get_var("BOOT_TO_SNAPSHOT"));
```
Comment (okurz) [inline tests/installation/grub_test.pm]: you need to include `use bootloader_setup 'boot_into_snapshot';`

**Comment by asmorodskyi** [comment]
This PR based on things which I descover while trying to merge https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/2996 .
@QingmingSu  probably I need your expertise here ( keep in mind that .grub_test_snapshot  used currently exclusively by  migration jobs ) 

Work is still in progress but I want to raise a discussion already to not miss something important and not waste time on something that will be thrown away after all

**Comment by QingmingSu** [comment]
@asmorodskyi The change looks good to me. The migration test won't be impacted by the change.

**Comment by mitiao** [comment]
Based on your comment:
> grub_test_snapshot used only in scenarios which have ROLLBACK_AFTER_MIGRATION

After refactor the block in sle/main.pm would no longer work:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/products/sle/main.pm#L1318
I think it can be removed if no any rollback test start on booting hdd image

File: tests/installation/grub_test.pm (hunk 1 above)
Comment (asmorodskyi) [inline tests/installation/grub_test.pm]: Done

**Comment by okurz** [comment]
@mitiao I am not sure what you are saying with "the block in sle/main.pm would no longer work". Is there a specific scenario with which @asmorodskyi could check his change?

@asmorodskyi verification run(s)? I know you did some as we worked on it together. Can you provide some? And I think it should be feasible if you just clone some, e.g. the "boot_to_snapshot" one we have in sle functional as well as a corresponding one for openSUSE - if there is any, if not, please remember it as a task for later - as well as migration jobs including the rollback after migration test steps.

**Comment by mitiao** [comment]
@okurz I mean if this refactor apply, following removed codes would not work properly:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/3011/commits/0a6d0ef45bcf629d38823caf30a6d8710fb31e0c

and this part worked for individual rollback test before like this:
https://openqa.suse.de/tests/818543

Since we have merged migration and rollback within one test, this part could be removed if nobody want to test this specific scenario.

**Comment by okurz** [comment]
@mitiao correct. @asmorodskyi can you remind why you delete that part?

**Comment by sysrich** [comment]
Would merge, but like @okurz asked for..verification runs?

**Comment by asmorodskyi** [comment]
proof run - http://kimball.arch.suse.de/tests/464 , still open for suggestions what actually I should check in this case

**Comment by asmorodskyi** [comment]
another proof run - http://kimball.arch.suse.de/tests/466

**Pull request merged**